In [2]:
import numpy as np
import pandas as pd

# 1

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [16]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])


Ковариация

In [21]:
cov_1= np.mean(zp*ks)-np.mean(zp)*np.mean(ks)
cov_1

9157.839999999997

In [25]:
cov_2 = np.cov(zp,ks,ddof=0)[0][1]
cov_2

9157.84

коэффициент корреляции Пирсона

In [30]:
cov_1/(np.std(zp)*np.std(ks))

0.8874900920739158

In [26]:
np.corrcoef(zp,ks)[0][1]

0.8874900920739162

In [28]:
df = pd.DataFrame({'zp' : zp, 'ks' : ks})
df

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [29]:
df.corr()

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

# 2

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [3]:
s=np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [9]:
s_avg=s.mean()

In [10]:
s_std = s.std(ddof=1)
s_std

10.54566788359614

In [11]:
s_avg+2.262*s_std/np.sqrt(10)

125.64339223691834

In [13]:
s_avg-2.262*s_std/np.sqrt(10)

110.55660776308164

Доверительный интервал [110.56;125.64]

# 3

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [14]:
174.2+1.96*5/np.sqrt(27)

176.08601087935276

In [15]:
174.2-1.96*5/np.sqrt(27)

172.31398912064722

Доверительный интервал [172.31;176.09]